Modules/Loadable/VesselEditor/Logic/vtkSlicerVesselEditorLogic.cxx

```diff
@@ -405,4 +405,36 @@ ::SplitTubeInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode, int tubeID, d
       break;
     }
   }
-}
\ No newline at end of file
+}
+
+//---------------------------------------------------------------------------
+void vtkSlicerVesselEditorLogic
+::RenumberTubesInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode)
+{
+  if (!spatialNode)
+  {
+    return ;
+  }
+
+  TubeNetType* spatialObject = spatialNode->GetSpatialObject();
+  char childName[] = "Tube";
+  TubeNetType::ChildrenListType* tubeList =
+    spatialObject->GetChildren(spatialObject->GetMaximumDepth(), childName);
+  int numberOfTubes = tubeList->size();
+  TubeIdType parentTubeId = numberOfTubes + 1000;
+  TubeIdType tubeIdIndex = 0;
+  for (TubeNetType::ChildrenListType::iterator tubeIt = tubeList->begin(); tubeIt != tubeList->end(); ++tubeIt)
+  {
+    VesselTubeType* currTube =
+      dynamic_cast<VesselTubeType*>((*tubeIt).GetPointer());
+    if (!currTube || currTube->GetNumberOfPoints() < 1)
+    {
+      continue;
+    }
+    currTube->SetId( tubeIdIndex );
+    currTube->SetParentId( parentTubeId );
```

Comment:
This is flattening the tree - giving all children and children's children, etc the same parent.

Perhaps use a recursive function - using only 1 depth of children.